### PR TITLE
[cattle-flu] consistent colours

### LINKED ISSUE #101: Consistent colours across all cattle-flu builds
Currently the (H5N1) cattle-flu builds use the [colour definitions of the H5N1 builds](https://github.com/nextstrain/avian-flu/blob/443e785b133ac84edb661fb85d745faff211e504/config/h5n1/colors_h5n1.tsv) (via [this config](https://github.com/nextstrain/avian-flu/blob/443e785b133ac84edb661fb85d745faff211e504/config/h5n1-cattle-outbreak.yaml#L42)) which are set for region, subtype, host, h5_label_clade and gisaid_clade. For the cattle-flu builds we want to define consistent colours primarily for "division", "division_metadata" and "data_source". Were applicable these should also be consistent across genome/segment builds. 

## Possible implementation for division

1. Define an ordering of ~50 states somewhere.
2. Have snakemake filter this by the values in the genome tree (just "division", as "division_metadata" will be a subset). Ideally we'd also filter to values appearing in the segment builds, but we can probably gloss over this and make the DAG much simpler.
3. Have snakemake generate a colour ramp for these values and write a new TSV which also includes the existing H5N1 definitions as  `augur export v2` only accepts one file for `--colors`. Duplicate these entries to also cover "division_metadata". 

## Possible implementation for division
This field isn't in H5N1 builds, so we could just add it to the underlying TSV. However it'd be cleaner to define this in a cattle-flu TSV and have snakemake add this in via step 3 above.

### PR DESCRIPTION
**Pull request by jameshadfield**
Closes #101. Final commit is a WIP and needs testing, but putting this up early for 👀 

Admin division (inferred):
image

Admin division (metadata):
image

Colors will always be internally consistent, but may change build to build as new values show up in the data.

(Aside: #100 introduced a new coloring for `division_metadata` but not a new geographic resolution, so we can colour by known metadata but will always have inferred values plotted on the map. That's quite informative actually, but subtle.)

### CODE REVIEW

**Comment by jameshadfield** [comment]
CI passed, trial builds passed ([ncbi](https://github.com/nextstrain/avian-flu/actions/runs/11618126992/job/32362010914), [fauna](https://github.com/nextstrain/avian-flu/actions/runs/11617767687)), results:
* [h5n1/ha/2y](https://nextstrain.org/staging/avian-flu/trials/james/consistent-colours/avian-flu/h5n1/ha/2y) - unchanged, as expected
* [h5n1-cattle-outbreak/genome](https://nextstrain.org/staging/avian-flu/trials/james/consistent-colours/avian-flu/h5n1-cattle-outbreak/genome?c=division) - as per screenshots above
* [h5n1-cattle-outbreak/ha](https://nextstrain.org/staging/avian-flu/trials/james/consistent-colours/avian-flu/h5n1-cattle-outbreak/ha?c=division) - uses same colours as /genome. Note the additional Missouri sample (`A/Missouri/121/2024`) which is grey as there's no Missouri data in the genome build

File: scripts/assign-colors.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    parser.add_argument('--duplications', type=str, required=False, nargs="+", metavar="categoryX=categoryY",
+                        help="Duplicate the colours generated for categoryX under the name categoryY")
+    return parser.parse_args()
+
+def read_ordering(fname):
+    # copied from <https://github.com/nextstrain/ncov/blob/80c06c15f0669c5990e0cc03628deeeedca7b720/scripts/assign-colors.py>
+    assignment = {}
+    with open(fname) as f:
+        for line in f.readlines():
+            array = line.lstrip().rstrip().split("\t")
```
Comment (genehack) [inline scripts/assign-colors.py:27]:
```suggestion
            array = line.strip().split("\t")
```

File: scripts/assign-colors.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    return assignment
+
+def read_schemes(fname):
+    # copied from <https://github.com/nextstrain/ncov/blob/80c06c15f0669c5990e0cc03628deeeedca7b720/scripts/assign-colors.py>
+    schemes = {}
+    counter = 0
+    with open(args.color_schemes) as f:
+        for line in f.readlines():
+            counter += 1
+            array = line.lstrip().rstrip().split("\t")
```
Comment (genehack) [inline scripts/assign-colors.py:44]:
```suggestion
            array = line.strip().split("\t")
```

**Review by genehack — APPROVED** [review]

File: scripts/assign-colors.py (hunk 1 above)
Comment (jameshadfield) [reply scripts/assign-colors.py:27]: Yes - better! I'm going to merge without changes however to keep it identical to where it was copied from and thus (maybe?) make [ENH: Add augur colors subcommand based on assign-colors.py scripts](https://github.com/nextstrain/augur/issues/1185) smoother when we get round to it.

**Pull request merged**